### PR TITLE
[zh-cn] init `BarcodeDetector`

### PR DESCRIPTION
**Pull request by skyclouds2001**
### Description

see https://developer.mozilla.org/en-US/docs/Web/API/BarcodeDetector

### CODE REVIEW

File: files/zh-cn/web/api/barcodedetector/index.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+## 构造方法
+
+- {{domxref('BarcodeDetector.BarcodeDetector', 'BarcodeDetector.BarcodeDetector()')}} {{Experimental_Inline}}
+  - : 创建并返回一个 `BarcodeDetector` 对象，带有可选的 `BarcodeDetectorOptions` 参数。
+
+## 静态方法
+
+- {{domxref('BarcodeDetector/getSupportedFormats_static', 'getSupportedFormats()')}} {{Experimental_Inline}}
+  - : 返回一个 {{jsxref('Promise')}}，它兑现一个 {{jsxref('Array')}}，包含受支持的[条形码格式类型](/zh-CN/docs/Web/API/Barcode_Detection_API#supported_barcode_formats)。
```
Comment (yin1999) [inline files/zh-cn/web/api/barcodedetector/index.md:20]:
```suggestion
  - : 返回一个 {{jsxref('Promise')}}，它兑现一个 {{jsxref('Array')}}，包含受支持的[条形码格式类型](/zh-CN/docs/Web/API/Barcode_Detection_API#支持的条形码格式)。
```

File: files/zh-cn/web/api/barcodedetector/index.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+---
+
+{{securecontext_header}}{{APIRef("Barcode Detector API")}}{{AvailableInWorkers}}{{SeeCompatTable}}
+
+{{domxref('Barcode Detection API', '', '', 'nocode')}} 的 **`BarcodeDetector`** 接口允许检测图像中的条形码和二维码。
+
+## 构造方法
+
+- {{domxref('BarcodeDetector.BarcodeDetector', 'BarcodeDetector.BarcodeDetector()')}} {{Experimental_Inline}}
+  - : 创建并返回一个 `BarcodeDetector` 对象，带有可选的 `BarcodeDetectorOptions` 参数。
```
Comment (yin1999) [inline files/zh-cn/web/api/barcodedetector/index.md:15]: How about this?

```suggestion
  - : 使用可选的 `BarcodeDetectorOptions` 参数创建并返回一个 `BarcodeDetector` 对象。
```

File: files/zh-cn/web/api/barcodedetector/index.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+## 实例方法
+
+- {{domxref('BarcodeDetector.detect', 'detect()')}} {{Experimental_Inline}}
+
+  - : 返回一个 {{jsxref('Promise')}}，它兑现一个具有以下属性的 `DetectedBarcode` 对象数组：
+
+    - `boundingBox`: 一个 {{domxref('DOMRectReadOnly')}}，返回表示检测到的条形码范围的矩形尺寸，与图像对齐。
+    - `cornerPoints`: 检测到的条形码的四个角点相对于图像的 x 和 y 坐标，从左上角开始顺时针旋转。由于图像内的透视变形，这可能不是方形的。
+    - `format`: 检测到的条形码格式。（有关格式的完整列表，请参阅\[API 概述页面]）
+    - `rawValue`: 一个从条形码数据解码的字符串。
```
Comment (yin1999) [inline files/zh-cn/web/api/barcodedetector/index.md:31]: As we have the API landing page for now, could you add the link back for "API 概述页面".

```suggestion
    - `boundingBox`: 一个 {{domxref('DOMRectReadOnly')}}，返回表示检测到的条形码范围的矩形尺寸，与图像对齐。
    - `cornerPoints`：检测到的条形码的四个角点相对于图像的 x 和 y 坐标，从左上角开始顺时针旋转。由于图像内的透视变形，这可能不是方形的。
    - `format`：检测到的条形码格式。（有关格式的完整列表，请参阅\[API 概述页面]）
    - `rawValue`：一个从条形码数据解码的字符串。
```

Comment (skyclouds2001) [reply files/zh-cn/web/api/barcodedetector/index.md]: see https://github.com/mdn/content/pull/31060

File: files/zh-cn/web/api/barcodedetector/index.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+## 实例方法
+
+- {{domxref('BarcodeDetector.detect', 'detect()')}} {{Experimental_Inline}}
+
+  - : 返回一个 {{jsxref('Promise')}}，它兑现一个具有以下属性的 `DetectedBarcode` 对象数组：
+
+    - `boundingBox`: 一个 {{domxref('DOMRectReadOnly')}}，返回表示检测到的条形码范围的矩形尺寸，与图像对齐。
+    - `cornerPoints`：检测到的条形码的四个角点相对于图像的 x 和 y 坐标，从左上角开始顺时针旋转。由于图像内的透视变形，这可能不是方形的。
+    - `format`：检测到的条形码格式。（有关格式的完整列表，请参阅\[API 概述页面]）
```
Comment (skyclouds2001) [inline files/zh-cn/web/api/barcodedetector/index.md:30]:
```suggestion
    - `format`：检测到的条形码格式。（有关格式的完整列表，请参阅[受支持的条形码格式类型](/zh-CN/docs/Web/API/Barcode_Detection_API#支持的条形码格式)）
```

File: files/zh-cn/web/api/barcodedetector/index.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
+---
+title: BarcodeDetector
+slug: Web/API/BarcodeDetector
+l10n:
+  sourceCommit: 5d6753b16efeb5b4d5cd91dcea1ec82f85b4ec21
```
Comment (skyclouds2001) [inline files/zh-cn/web/api/barcodedetector/index.md:5]:
```suggestion
  sourceCommit: 6bfd45572529a34b4e3a059dcdd37ca6def2e0cd
```

**Review by yin1999 — APPROVED** [review]
LGTM. Thank you @skyclouds2001

**Pull request merged**